druid/druid.sh

```diff
@@ -0,0 +1,538 @@
+    processing_buffer=1073741824
+  fi
+  cores=$(grep -c ^processor /proc/cpuinfo)
+  cores=$((${cores}-1))
+  num_threads=$(max 1 ${cores})
+  if [[ ${num_threads} < 4 ]];then
+    num_merge_buff=2
+  else
+    num_merge_buff=$(max 2 ${num_threads}/4)
+  fi
+  xmx=$(calculate_xmx ${processing_buffer} ${num_threads} ${num_merge_buff})
+  xx=$((${num_threads} * ${processing_buffer} * 4))
+  cat << EOF > conf/druid/${service_name}/jvm.config
+-XX:MaxDirectMemorySize=${xx}
+-Xmx${xmx}
+-Duser.timezone=UTC
+-Dfile.encoding=UTF-8
+-Djava.io.tmpdir=var/tmp
+-Djava.util.logging.manager=org.apache.logging.log4j.jul.LogManager
+-Dderby.stream.error.file=var/druid/derby.log
+EOF
+}
+
+function configure_druid() {
+  local java_home
+  local hadoop_conf_dir
+  local druid_overlord_port
+  local druid_coordinator_port
+  local zookeeper_client_port
+  local zookeeper_list
+  local tmp_dir
+  local cluster_name
+  local sql_host
+  local master_additional
+  local hive_metastore_instance
+  local gcs_bucket
+  local base_extensions
+  local gcs_enabled_extensions
+
+  base_extensions="[\"druid-kafka-eight\", \"druid-histogram\", \"druid-datasketches\", \"druid-lookups-cached-global\", \"mysql-metadata-storage\", \"druid-hdfs-storage\",\"druid-kafka-indexing-service\"]"
+  gcs_enabled_extensions="[\"druid-kafka-eight\", \"druid-histogram\", \"druid-datasketches\", \"druid-lookups-cached-global\", \"mysql-metadata-storage\", \"druid-hdfs-storage\", \"druid-kafka-indexing-service\", \"druid-google-extensions\"]"
+  master_additional=$(/usr/share/google/get_metadata_value attributes/dataproc-master-additional)
+  cluster_name=$(/usr/share/google/get_metadata_value attributes/dataproc-cluster-name)
+  hive_metastore_instance=$(/usr/share/google/get_metadata_value attributes/hive-metastore-instance)
+  gcs_bucket=$(/usr/share/google/get_metadata_value attributes/gcs-bucket)
+
+  # Set localhost as sql host for cloud-sql-proxy or master node for other configurations
+  if [[ ${hive_metastore_instance} == "" ]]; then
+    if [ "${master_additional}" == '' ]; then
+      sql_host="${cluster_name}-m"
+    else
+      sql_host="${cluster_name}-m-0"
+    fi
+  else
+    sql_host='localhost'
+  fi
+
+  druid_overlord_port=$(/usr/share/google/get_metadata_value attributes/druid-overlord-port)
+  druid_coordinator_port=$(/usr/share/google/get_metadata_value attributes/druid-coordinator-port)
+  java_home=$(echo ${JAVA_HOME})
+  hadoop_conf_dir="/etc/hadoop/conf"
+
+  zookeeper_client_port=$(grep 'clientPort' /etc/zookeeper/conf/zoo.cfg \
+    | tail -n 1 \
+    | cut -d '=' -f 2)
+
+  zookeeper_list=$(grep '^server\.' /etc/zookeeper/conf/zoo.cfg \
+    | cut -d '=' -f 2 \
+    | cut -d ':' -f 1 \
+    | sort \
+    | uniq \
+    | sed "s/$/:${zookeeper_client_port}/" \
+    | xargs echo  \
+    | sed "s/ /,/g")
+
+  cd ${DRUID_DIR}
+
+  ln -sf /usr/share/java/mysql-connector-java.jar extensions/mysql-metadata-storage/mysql-connector-java-5.1.38.jar
+
+  java \
```

Comment:
These two calls to `org.apache.druid.cli.Main tools pull-deps` can be folded together in one call.

java
 ...
 -c "org.apache.druid.extensions:mysql-metadata-storage:${DRUID_VERSION}" \
-c "org.apache.druid.extensions:druid-hdfs-storage:${DRUID_VERSION}"

We should also consider following: 
-c "org.apache.druid.extensions:druid-hdfs-storage:${DRUID_VERSION}" \
    -c "org.apache.druid.extensions:mysql-metadata-storage:${DRUID_VERSION}" \
    -c "org.apache.druid.extensions:druid-kafka-eight:${DRUID_VERSION}" \
    -c "org.apache.druid.extensions.contrib:druid-google-extensions:${DRUID_VERSION}" \
    -c "org.apache.druid.extensions.contrib:druid-distinctcount:${DRUID_VERSION}"

Reply:
@aniket486  I simplified this as you suggested.

Reply:
It's already made with this:
`  java \
    -cp "lib/*" \
    -Ddruid.extensions.directory="extensions" \
    -Ddruid.extensions.hadoopDependenciesDir="hadoop-dependencies" \
    org.apache.druid.cli.Main tools pull-deps \
    --no-default-hadoop \
    -c "org.apache.druid.extensions:druid-hdfs-storage:${DRUID_VERSION}" \
    -c "org.apache.druid.extensions:mysql-metadata-storage:${DRUID_VERSION}" \
    -c "org.apache.druid.extensions:druid-kafka-eight:${DRUID_VERSION}" \
    -c "org.apache.druid.extensions.contrib:druid-google-extensions:${DRUID_VERSION}" \
    -c "org.apache.druid.extensions.contrib:druid-distinctcount:${DRUID_VERSION}"`